server/src/services/asset-media.service.ts

```diff
@@ -208,7 +213,7 @@ export class AssetMediaService extends BaseService {
     const asset = await this.findOrFail(id);
     const size = dto.size ?? AssetMediaSize.THUMBNAIL;
 
-    const { thumbnailFile, previewFile, fullsizeFile } = getAssetFiles(asset.files);
+    const { thumbnailFile, previewFile, fullsizeFile } = getAssetFiles(asset.files ?? []);
```

Comment:
It is probably worth splitting this file stuff out into a separate PR. Here, in particular the could use something (instead of `findOrFail` to only get the asset's id, file paths, ownerId, etc. We only need a few fields to implement this method.